server/app/mutations/grid_certificates/get_certificate.rb

```diff
@@ -54,10 +57,8 @@ def execute
           end
         end
 
-        Timeout::timeout(30) {
-          info 'waiting for DNS validation...'
-          sleep 1 until challenge.verify_status == 'valid'
-          info 'DNS validation complete'
+        wait_until!("domain verification for #{domain} is valid", interval: 1, timeout: 30, threshold: 10) {
+          challenge.verify_status == 'valid'
```

Comment:
Separate issue with the existing code: #2688

It's just way more likely to fail here compared to the old `dns-01` validation, because the server would itself also validate the DNS challenge record, which means that the domain authorization challenge is unlikely to fail.

However, particularly with the delayed `LE_TLS_SNI_*` challenge cert deployment from the `GridCertificates::AuthorizeDomain`, and the lack of our own client-side validation for the TLS-SNI challenge, this is now far more likely to fail here.